### PR TITLE
logwriter: Avoid null pointer access from suspended writer

### PR DESCRIPTION
**Pull request by MrAnno**
I've accidentally closed #2645 with a stale push.

Original author: @kyeongy

> There is a chance with rare timing that a suspended writer tries to
access already freed client, which causes a SEGV.

https://github.com/balabit/syslog-ng/pull/2645#issuecomment-477816858
https://github.com/balabit/syslog-ng/pull/2645#issuecomment-500114948
https://github.com/balabit/syslog-ng/pull/2645#issuecomment-500254227
https://github.com/balabit/syslog-ng/pull/2645#issuecomment-500420282

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
@kira-syslogng do perftest

**Comment by kira-syslogng** [comment]
Kira-performance-test: Build SUCCESS

**Comment by MrAnno** [comment]
@kyeongy Thanks again for catching and fixing this crash!

File: lib/logwriter.c [hunk 1]
```diff
@@ -547,6 +545,8 @@ log_writer_stop_watches(LogWriter *self)
 
       self->watches_running = FALSE;
     }
+
+  log_writer_stop_suspend_timer(self);
```
Comment (Kokan) [inline lib/logwriter.c]: Before in the `log_writer_io_handler` did not called `log_writer_stop_suspend_timer` unconditional. Just leave the note, as every other missing place, you first introduced the `log_writer_stop_suspend_timer`.

But I think it is correct to stop the `suspend_timer` there, and even required.

**Pull request merged**